src/components/FormName.vue

```diff
@@ -0,0 +1,230 @@
+<template>
+  <form
+    class="form-wrapper"
+    @submit.prevent="onSubmit"
+  >
+    <BaseInput
+      type="text"
+      label="first name"
+      placeholder="e.g. Jessica"
+      v-model="firstName"
+      @blur="isFirstNameTouched = true"
+    />
+    <div class="error">{{ firstNameError }}</div>
+
+    <BaseInput
+      type="text"
+      label="last name"
+      placeholder="e.g. Walton"
+      v-model="lastName"
+      @blur="isLastNameTouched = true"
+    />
+    <div class="error">{{ lastNameError }}</div>
+
+    <BaseInput
+      type="date"
+      label="date of birth"
+      placeholder="DD/MM/YYYY"
+      v-model="birthDate"
+      @blur="isBirthDateTouched = true"
+    />
+    <div class="error">{{ birthDateError }}</div>
+    <p class="hint">You should be minimum 18 years old</p>
+
+    <div class="checkbox">
+      <input
+        type="checkbox"
+        id="checkbox"
+        v-model="checked"
+      />
+      <label for="checkbox">I accept Privacy Policy</label>
+
+    </div>
+    <div class="form-wrapper__action">
+      <router-link :to="{ name: 'Login' }">
+        <BaseButton
+          caption="Log in instead"
+          buttonType="secondary"
+        />
+      </router-link>
+      <button
+        type="submit"
+        class="form-wrapper__button"
+        :disabled="!isFormValid"
+      >
+        Register
+      </button>
+    </div>
+  </form>
+</template>
+
+<script>
+/* eslint-disable */
+import BaseInput from "./BaseInput.vue";
+import BaseButton from "./BaseButton.vue";
+export default {
+  name: "FormLogin",
+  components: {
+    BaseInput,
+    BaseButton,
+  },
+  data() {
+    return {
+      firstName: "",
+      lastName: "",
+      birthDate: "",
+      checked: false,
+      isFirstNameTouched: false,
+      isLastNameTouched: false,
+      isBirthDateTouched: false,
+    };
+  },
+  computed: {
+    firstNameError() {
+      if (!this.isFirstNameTouched) {
+        return "";
+      }
+      if (!this.firstName) {
+        return "First name is required!";
+      }
+    },
+
+    lastNameError() {
+      if (!this.isLastNameTouched) {
+        return "";
+      }
+      if (!this.lastName) {
+        return "Last name is required!";
+      }
+    },
+
+    calculateAge() {
+      const today = new Date();
+      const birthDate = new Date(this.birthDate);
+      const age = today.getFullYear() - birthDate.getFullYear();
+      if (age >= 18) {
```

Comment:
To make this if/else statement a little bit shorter you can use [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)
So it should look like this:
```
return age >= 18 ? true : false
```

Reply:
and even shorter but still the same effect 🙈 
```
return age >= 18
```